scripts/util/snapstore.py

```diff
@@ -11,20 +13,55 @@
     "Snap-Device-Series": "16",
     "User-Agent": "Mozilla/5.0",
 }
+TIMEOUT = 10
 
 
 def info(snap_name):
-    req = Request(INFO_URL + snap_name, headers=HEADERS)
-    try:
-        with urlopen(req) as response:  # nosec
-            return json.loads(response.read().decode())
-    except HTTPError as e:
-        LOG.exception("HTTPError ({%s}): {%s} {%s}", req.full_url, e.code, e.reason)
-        raise
-    except URLError as e:
-        LOG.exception("URLError ({%s}): {%s}", req.full_url, e.reason)
-        raise
+    r = requests.get(INFO_URL + snap_name, headers=HEADERS, timeout=TIMEOUT)
+    r.raise_for_status()
+    return json.loads(r.text)
+
+
+def track_exists(snap_name: str, track_name: str):
+    snap_info = info(snap_name)
+    for channel_data in snap_info["channel-map"]:
+        track = channel_data["channel"]["track"]
+        if track == track_name:
+            return True
+    return False
 
 
 def ensure_track(snap_name: str, track_name: str):
-    pass
+    LOG.info("Ensuring track: %s %s", snap_name, track_name)
+    if not track_exists(snap_name, track_name):
+        create_track(snap_name, track_name)
+    else:
+        LOG.info("Track already exists: %s %s", snap_name, track_name)
+
+
+def create_track(snap_name: str, track_name: str):
```

Comment:
nit: Maybe we can add a doc string? And also:
```suggestion
def create_track(snap_name: str, track_name: str) -> None:
```